### PR TITLE
Update Org VDC Networks

### PR DESCRIPTION
**Pull request by dataclouder**
Adds the ability of updating Org VDC networks in place. Until now, we could only remove and re-create them.

Added methods `orgVdcNetwork.Update`, `orgVdcNetwork.UpdateAsync`, and `OrgVdcNetwork.Rename`

### CODE REVIEW

File: types/v56/types.go [hunk 1]
```diff
@@ -212,14 +212,14 @@ type NetworkConfiguration struct {
...
 	Features                       *NetworkFeatures `xml:"Features,omitempty"`
-	GuestVlanAllowed               *bool            `xml:"GuestVlanAllowed,omitempty"`
 
 	// SubInterface and DistributedInterface are mutually exclusive
 	// When they are both nil, it means the "internal" interface (the default) will be used.
 	// When one of them is set, the corresponding interface will be used.
 	// They cannot be both set (we'll get an API error if we do).
 	SubInterface         *bool `xml:"SubInterface,omitempty"`
 	DistributedInterface *bool `xml:"DistributedInterface,omitempty"`
+	GuestVlanAllowed     *bool `xml:"GuestVlanAllowed,omitempty"`
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply types/v56/types.go]: I haven't come across any issues with order after this change.

File: govcd/edgegateway_test.go [hunk 2]
```diff
@@ -743,3 +743,51 @@ func (vcd *TestVCD) TestListEdgeGateway(check *C) {
...
+	newName := "UpdatedEGWName"
+	newDescription := "Updated description"
+
+	edge.EdgeGateway.Name = newName
+	edge.EdgeGateway.Description = newDescription
+
+	err = edge.Update()
+	check.Assert(err, IsNil)
+
+	edge, err = vcd.vdc.GetEdgeGatewayById(saveEGW.ID, true)
```
Comment (vbauzysvmware) [inline govcd/edgegateway_test.go]: I think `update` should also update referenced Instance, not only send data.
So `edge.Update` -> would means that after `edge.EdgeGateway.Name` I would see new name. I don't need 
`edge, err = vcd.vdc.GetEdgeGatewayById(saveEGW.ID, true)`
Or return updated instance - usually we do that.

File: govcd/orgvdcnetwork.go [hunk 3]
```diff
@@ -236,3 +236,111 @@ func (vdc *Vdc) FindEdgeGatewayNameByNetwork(networkName string) (string, error)
...
+	}
+
+	return vdc.GetEdgeGatewayByName(edgeGatewayName, false)
+}
+
+// UpdateAsync will change the contents of a network using the information in the
+// receiver data structure.
+func (orgVdcNet *OrgVDCNetwork) UpdateAsync() (Task, error) {
+	if orgVdcNet.OrgVDCNetwork.HREF == "" {
+		return Task{}, fmt.Errorf("cannot update network: HREF is empty")
```
Comment (vbauzysvmware) [inline govcd/orgvdcnetwork.go]: We should update all errors to be more specific-> "cannot update Org VDC network: HREF is empty"

File: govcd/orgvdcnetwork_test.go [hunk 4]
```diff
@@ -151,6 +159,34 @@ func (vcd *TestVCD) testCreateOrgVdcNetworkRouted(check *C, ipSubnet string, sub
...
+	network.OrgVDCNetwork.Configuration.IPScopes.IPScope[0].IPRanges.IPRange[0].EndAddress = updatedEndAddress
+
+	err = network.Update()
+	check.Assert(err, IsNil)
+	AddToCleanupList(updatedNetworkName,
+		"network",
+		vcd.org.Org.Name+"|"+vcd.vdc.Vdc.Name,
+		"Test_CreateOrgVdcNetworkRouted")
+
+	network, err = vcd.vdc.GetOrgVdcNetworkById(networkId, true)
```
Comment (vbauzysvmware) [inline govcd/orgvdcnetwork_test.go]: Similar comment here regarding updating referenced instance during update.

File: types/v56/types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply types/v56/types.go]: I checked, at least we don't need to handle GuestVlanAllowed between versions. 9.1 already supports it.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,9 @@
-## 2.6.0 (Unreleased)
+## 2.7.0 (Unreleased)
+
+* Added methods `OrgVdcNetwork.Update`, `OrgVdcNetwork.UpdateAsync`, and `OrgVdcNetwork.Rename`
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply CHANGELOG.md]: done

File: govcd/edgegateway.go [hunk 6]
```diff
@@ -1398,3 +1404,24 @@ func getNetworkNameAndTypeByVnicIndex(vNicIndex int, vnics *types.EdgeGatewayInt
 
 	return networkName, networkType, nil
 }
+
+func (egw *EdgeGateway) UpdateAsync() (Task, error) {
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply govcd/edgegateway.go]: added

File: govcd/edgegateway_test.go (hunk 2 above)
Comment (dataclouder) [reply govcd/edgegateway_test.go]: Added explicit refresh at the end of Update, as agreed

File: govcd/orgvdcnetwork.go (hunk 3 above)
Comment (dataclouder) [reply govcd/orgvdcnetwork.go]: updated

File: govcd/orgvdcnetwork_test.go (hunk 4 above)
Comment (dataclouder) [reply govcd/orgvdcnetwork_test.go]: addressed

**Comment by dataclouder** [comment]
> LGTM, I would only like to ask to extend the PR description (and commit message) to include narrative what was not possible before and what the additions allow to do now more specifically (from the perspective of casual user who will be reading this PR description after clicking link in the changelog after referencing it with the next release).

Previously, we could not update networks: we had to remove and re-create them. That's what the Update functions are about

**Pull request merged**